nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,677 @@
+
+    Streamlines of a tractogram can be in any coordinate system of your
+    choice as long as you provide the correct `affine_to_rasmm` matrix, at
+    construction time, that brings the streamlines back to *RAS+*, *mm* space,
+    where the coordinates (0,0,0) corresponds to the center of the voxel
+    (opposed to a corner).
+
+    Attributes
+    ----------
+    streamlines : :class:`ArraySequence` object
+        Sequence of $T$ streamlines. Each streamline is an ndarray of
+        shape ($N_t$, 3) where $N_t$ is the number of points of
+        streamline $t$.
+    data_per_streamline : dict of 2D arrays
+        Dictionary where the items are (str, 2D array).
+        Each key represents an information $i$ to be kept along side every
+        streamline, and its associated value is a 2D array of shape
+        ($T$, $P_i$) where $T$ is the number of streamlines and $P_i$ is
+        the number scalar values to store for that particular information $i$.
+    data_per_point : dict of :class:`ArraySequence` objects
+        Dictionary where the items are (str, :class:`ArraySequence`).
+        Each key represents an information $i$ to be kept along side every
+        point of every streamline, and its associated value is an iterable
+        of ndarrays of shape ($N_t$, $M_i$) where $N_t$ is the number of
+        points for a particular streamline $t$ and $M_i$ is the number
+        scalar values to store for that particular information $i$.
+    """
+    def __init__(self, streamlines=None,
+                 data_per_streamline=None,
+                 data_per_point=None,
+                 affine_to_rasmm=np.eye(4)):
+        """
+        Parameters
+        ----------
+        streamlines : iterable of ndarrays or :class:`ArraySequence`, optional
+            Sequence of $T$ streamlines. Each streamline is an ndarray of
+            shape ($N_t$, 3) where $N_t$ is the number of points of
+            streamline $t$.
+        data_per_streamline : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            streamline, and its associated value is an iterable of ndarrays of
+            shape ($P_i$,) where $P_i$ is the number scalar values to store
+            for that particular information $i$.
+        data_per_point : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            point of every streamline, and its associated value is an iterable
+            of ndarrays of shape ($N_t$, $M_i$) where $N_t$ is the number of
+            points for a particular streamline $t$ and $M_i$ is the number
+            scalar values to store for that particular information $i$.
+        affine_to_rasmm : ndarray of shape (4, 4)
```

Comment:
It is a good idea though and it would help us dealing with streamlines coming from file formats without information about the space they are in. However, allowing `None` would break a lot of things :-/ and I think it should be handled in another PR (before adding support to other file formats). For now, I will leave the default to be the identity as I think it is more convenient for the user.